hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoUpdateTransitionLogic.java

```diff
@@ -70,120 +62,79 @@
 public class CryptoUpdateTransitionLogic implements TransitionLogic {
 	private static final Logger log = LogManager.getLogger(CryptoUpdateTransitionLogic.class);
 
-	private static final EnumSet<AccountProperty> EXPIRY_ONLY = EnumSet.of(AccountProperty.EXPIRY);
-
 	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
 
-	private final HederaLedger ledger;
+	private final AccountStore accountStore;
 	private final OptionValidator validator;
 	private final TransactionContext txnCtx;
+	private final TransactionRecordService transactionRecordService;
 	private final GlobalDynamicProperties dynamicProperties;
 
 	@Inject
 	public CryptoUpdateTransitionLogic(
-			HederaLedger ledger,
+			AccountStore accountStore,
```

Comment:
nit: I think that we would benefit from a proper ordering of the arguments in the constructor, where the first arguments are transaction body-related, followed by dependencies, serving the transaction logic itself like stores, validators, etc.